### PR TITLE
INTMDB-459 Cloud Provider Access

### PR DESCRIPTION
**Pull request by Adelmar92**
## Description

Added new resource CLOUD PROVIDER ACCESS

Returns, adds, authorizes, and removes AWS IAM roles in Atlas.

How to test:

Can be manually executed by using this json inputs and the command:

`cfn invoke resource {CREATE | READ | UPDATE | DELETE | LIST} file-name.json`
    
CREATE
```json
{
  "desiredResourceState": {
    "ProjectId": {PROJECT-ID},
    "ApiKeys": {
          "PublicKey": {PUBLIC-KEY},
          "PrivateKey": {PRIVATE-KEY}
    }
  },
  "previousResourceState": {
  }
}
```
    
READ
```json
{
  "desiredResourceState": {
    "ProjectId": {PROJECT-ID},
    "ApiKeys": {
          "PublicKey": {PUBLIC-KEY},
          "PrivateKey": {PRIVATE-KEY}
    }
  },
  "previousResourceState": {
  }
}
```

DELETE
```json
{
  "desiredResourceState": {
    "ProjectId": {PROJECT-ID},
    "RequestId": {RequestId},
    "RoleId": {ROLE-ID},
    "ApiKeys": {
          "PublicKey": {PUBLIC-KEY},
          "PrivateKey": {PRIVATE-KEY}
    }
  },
  "previousResourceState": {
  }
}
```

LIST
```json
{
  "desiredResourceState": {
    "ProjectId": {PROJECT-ID},
    "RequestId": {RequestId},
    "ApiKeys": {
          "PublicKey": {PUBLIC-KEY},
          "PrivateKey": {PRIVATE-KEY}
    }
  },
  "previousResourceState": {
  }
}
```

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by Zuhairahmed** [comment]
@Adelmar92 can you resolve linting error and branch conflicts on this PR?

File: cfn-resources/cloud-provider-access/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Mongodb::Atlas::CloudProviderAccess
+
+Congratulations on starting development!
```
Comment (Zuhairahmed) [inline cfn-resources/cloud-provider-access/README.md:3]: Can you copy format from other resources for README file? This looks off...

File: examples/cloud-provider-access/cloudprovideraccess.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  "AWSTemplateFormatVersion": "2010-09-09",
+  "Description": "MongoDB Atlas AWS CloudFormation Example for MongoDB::Atlas::ThirdPartyIntegration",
```
Comment (Zuhairahmed) [inline examples/cloud-provider-access/cloudprovideraccess.json:3]:
```suggestion
  "Description": "MongoDB Atlas AWS CloudFormation Example for MongoDB::Atlas::CloudProviderAccess",
```

**Review by Zuhairahmed — CHANGES_REQUESTED** [review]
Few comments, please address. Otherwise good start @Adelmar92!

File: cfn-resources/cloud-provider-access/README.md (hunk 1 above)
Comment (Adelmar92) [reply cfn-resources/cloud-provider-access/README.md:3]: @parthasarathy-varadhan you had a readme file you told me you want to add?

**Review by govardhanpagidi — APPROVED** [review]
LGTM!

File: cfn-resources/cloud-provider-access/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+
+```bash
+repo_root=$(git rev-parse --show-toplevel)
+source <(${repo_root}/quickstart-mongodb-atlas/scripts/export-mongocli-config.py)
+${repo_root}/quickstart-mongodb-atlas/scripts/launch-x-quickstart.sh ${repo_root}/cfn-resources/cloud-provider-access/test/cloudprovideraccess.sample-template.yaml SampleProject1 ParameterKey=OrgId,ParameterValue=${ATLAS_ORG_ID}
+```
+
+## For More Information
+See the MongoDB Atlas API [Cloud-Provider-Access](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Cloud-Provider-Access) Documentation.
```
Comment (Zuhairahmed) [inline cfn-resources/cloud-provider-access/README.md:65]:
```suggestion
See the MongoDB Atlas API Endpoint [Cloud-Provider-Access](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Cloud-Provider-Access) Documentation.
```

**Review by Zuhairahmed — APPROVED** [review]
Few more comments, please address. Not blocking. Otherwise LGTM!

**Pull request merged**